### ISSUE TITLE
[Bug][Expandable] Will make other expandable element which in front of this Attirbute not update 

### ISSUE THREAD

**Issue by wojiuxuihuan** [dbrizov/NaughtyAttributes#294]
Here's the code that reproduce the bug:
```C#
    public class ExpandableTest : MonoBehaviour
    {
        public UnityEvent onClick;//Bug: the List can't be modify
        [ResizableTextArea]
        public string multiLine;//Bug: the content can't be save after hit 'Enter'
        [Expandable]
        public ScriptableObject obj0;
    }

```

When you edit the onClick/multiLine field in Inspector, you will find that their value will not get save:
![image](https://user-images.githubusercontent.com/13210990/147955242-d70a0711-86ad-4e31-b7e1-c437b0803963.png)

But if you move the obj0 on top of the onClick/multiLine field, they will work again.

Unity Version: 2021.2.7f1c1

**Comment by rhys-vdw** [comment]
More generally this affects any field that precedes a field decorated with `ExpandableAttribute`, including `int` etc, regardless of whether they have NA decorators. The field can be edited but will revert to its previous value when blurred.

Collapsing/expanding the field or removing its asset reference makes no difference.

This is true in two of my projects using NaughtyAttributes, both using Unity 2020.3.21.

**Comment by rhys-vdw** [comment]
This issue is present in 2.1.1 only. Rolling back to 2.1.0 resolves the problem (but presumably will then suffer from #232). @dbrizov it seems that the issue was introduced in e4edd04d78aeb27cb99134b8fc84f521485de6de.

**Comment by rhys-vdw** [comment]
I have discovered that if you comment out this line the issue goes away.

https://github.com/rhys-vdw/NaughtyAttributes/blob/77b2b0e2e287c7ead3c5c1a9eee4a8e621711e05/Assets/NaughtyAttributes/Scripts/Editor/PropertyDrawers/ExpandablePropertyDrawer.cs#L74

My guess is that when the drawer is called on the blur event this code will somehow reset the value on the main inspector object before it has a chance to commit the changes via the final call to `ApplyModifiedProperties` here:

https://github.com/rhys-vdw/NaughtyAttributes/blob/77b2b0e2e287c7ead3c5c1a9eee4a8e621711e05/Assets/NaughtyAttributes/Scripts/Editor/NaughtyInspector.cs#L133

Why this is affecting the main object is not clear to me though, presumably some subtlety of other changes in the problematic commit.

**Comment by rhys-vdw** [comment]
Why was this closed? #295 is still not merged, was there another fix?

**Comment by wojiuxuihuan** [comment]
> Why was this closed? #295 is still not merged, was there another fix?

I see the author had reply your pull request, and I think it might not well for two issues link to the same problem so this one is closed.

**Comment by rhys-vdw** [comment]
There is no other issue @wojiuxuihuan. This should stay open until the PR is merged, at which point it will automatically be closed by GitHub.

**Comment by wojiuxuihuan** [comment]
> There is no other issue @wojiuxuihuan. This should stay open until the PR is merged, at which point it will automatically be closed by GitHub.

I see, thanks for notify!

**Comment by dbrizov** [comment]
#295 PR merged into the **v2** branch 7cddf69bca9d09eb8205498ebaf6bdc50f7641fd
